src/input/datetime.rs

```diff
@@ -524,17 +535,43 @@ impl TzInfo {
         None
     }
 
+    fn fromutc<'py>(&self, dt: &'py PyDateTime) -> PyResult<&'py PyAny> {
+        let py = dt.py();
+        dt.call_method1("__add__", (self.utcoffset(py, py.None().as_ref(py))?,))
+    }
+
     fn __repr__(&self) -> String {
         format!("TzInfo({})", self.__str__())
     }
 
     fn __str__(&self) -> String {
         if self.seconds == 0 {
-            "UTC".to_string()
-        } else {
-            let mins = self.seconds / 60;
-            format!("{:+03}:{:02}", mins / 60, (mins % 60).abs())
+            return "UTC".to_string();
+        }
+
+        let (mins, seconds) = (self.seconds / 60, self.seconds % 60);
+        let mut result = format!(
+            "{}{:02}:{:02}",
+            if self.seconds.signum() >= 0 { "+" } else { "-" },
+            (mins / 60).abs(),
+            (mins % 60).abs()
+        );
+
+        if seconds != 0 {
+            result.push_str(&format!(":{:02}", seconds.abs()));
         }
+
+        result
+    }
+
+    fn __hash__(&self) -> u64 {
```

Comment:
@davidhewitt is this the best way to hash `TzInfo`?

Reply:
at least it's the pyo3 docs way:)

https://pyo3.rs/main/class/object.html#hashing

Reply:
I think this is fine.